### PR TITLE
ci: split windows in mandatory and ext_win stages

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Split the Windows OS build/test stages in two meta-stages:
- Mandatory for `windows-2022` and `windows-2016`. This is mandatory for all the builds in the CI (PRs, branches, tags)
- Extended support for other windows versions. This is only available for branches or UI interactions with the parameter `runAllStages`)

`Windows-2019` is not anymore the default OS but `windows-2022` and `windows-2016`

Removed tag event builds, every commit in this project is validated in the CI, so there is no need to run a build for tags.

Move the `arm` stages to mandatory.

Mandatory stages run always. But extended support for other windows versions only run for branches.

## Why is it important?

Faster builds on a PR basis since the extended meta-stage has been simplified a bit by moving some stages to the `mandatory` meta-stage or the new `extended-win` meta-stage.

In other words, the existing meta-stages were:

`checkout` - `lint` - `mandatory` - `extended` - `package`

and now:

`checkout` - `lint` - `mandatory` - `extended` - `extended-win` - `package`

and each meta-stage runs in parallel a set of stages.

## Questions

- [ ] Is it required to run on Windows for each Beat? Or are there any Beats that are not windows supported? If so which should be disabled?

### CODE REVIEW

**Comment by ph** [comment]
>  Is it required to run on Windows for each Beat? Or are there any Beats that are not windows supported? If so which should be disabled?

I would say dockerlogbeat which isn't Windows, I don't know exactly the states or what test is run for that component.
Everything else need windows testing.

**Review by ph — APPROVED** [review]
I am not completely familiar with everything Jenkins pipeline related,
I've looked at the current ci job run for this PR and the changes in the YML and they do appear fine to me. Let's have more eye on this.

File: x-pack/auditbeat/Jenkinsfile.yml [hunk 1]
```diff
@@ -102,7 +97,6 @@ stages:
         stage: packaging
         when:
             branches: false    ## Only on a PR basis for the time being
-            tags: false        ## packaging on branches/tags is already in place with the downstream build.
```
Comment (cachedout) [inline x-pack/auditbeat/Jenkinsfile.yml:105]: Nit: In the future, I think it would be better to split these kind of changes out into a separate PR. I think we mixed two very different concerns in this single PR.

File: x-pack/filebeat/Jenkinsfile.yml [hunk 2]
```diff
@@ -64,38 +62,37 @@ stages:
                 - "macOS"
             parameters:
                 - "macosTest"
-            tags: true         ## for all the tags
         stage: extended
-    windows:
+    windows-2022:
```
Comment (cachedout) [inline x-pack/filebeat/Jenkinsfile.yml:66]: I am a bit curious about the change here from `windows` -> `windows-2022`. Since all the other top-level items are specific to a release (i.e. `windows-2019` like on L71), I am curious why there was a generic entry for `windows` to begin with. Was it some kind of default or catch-all, or was it just simply named this way for historical reasons? (Hopefully this question makes sense.)

File: x-pack/auditbeat/Jenkinsfile.yml (hunk 1 above)
Comment (v1v) [reply x-pack/auditbeat/Jenkinsfile.yml:105]: Indeed, Let me remove it to help with the review

Comment (v1v) [reply x-pack/auditbeat/Jenkinsfile.yml]: I reverted those changes now

File: x-pack/filebeat/Jenkinsfile.yml (hunk 2 above)
Comment (cachedout) [reply x-pack/filebeat/Jenkinsfile.yml:69]: That's perfectly clear. Thank you for giving me the history!

**Comment by v1v** [comment]
@mergifyio backport 7.17
